src/components/VegaGraph.tsx

```diff
@@ -0,0 +1,448 @@
+          "padding": 0.05,
+          "round": true
+        },
+        {
+          "name": "yscale",
+          "domain": {"data": "table", "field": "amount"},
+          "nice": true,
+          "range": "height"
+        }
+      ],    
+      "axes": [
+        { "orient": "bottom", "scale": "xscale" },
+        { "orient": "left", "scale": "yscale" }
+      ],    
+      "marks": [
+        {
+          "type": "rect",
+          "from": {"data":"table"},
+          "encode": {
+            "enter": {
+              "x": {"scale": "xscale", "field": "category"},
+              "width": {"scale": "xscale", "band": 1},
+              "y": {"scale": "yscale", "field": "amount"},
+              "y2": {"scale": "yscale", "value": 0}
+            },
+            "update": {
+              "fill": {"value": "steelblue"}
+            },
+            "hover": {
+              "fill": {"value": "red"}
+            }
+          }
+        },
+        {
+          "type": "text",
+          "encode": {
+            "enter": {
+              "align": {"value": "center"},
+              "baseline": {"value": "bottom"},
+              "fill": {"value": "#333"}
+            },
+            "update": {
+              "x": {"scale": "xscale", "signal": "tooltip.category", "band": 0.5},
+              "y": {"scale": "yscale", "signal": "tooltip.amount", "offset": -2},
+              "text": {"signal": "tooltip.amount"},
+              "fillOpacity": [
+                {"test": "datum === tooltip", "value": 0},
+                {"value": 1}
+              ]
+            }
+          }
+        }
+      ]
+    }    
+    const runtime = parse(specs);    
+    const graphVega = document.getElementById('graphVega');
+    var viewGraph = new View(runtime)
+                .renderer('svg')     // set render type (defaults to 'canvas')
+                .initialize(graphVega);
+    viewGraph.run();
+  };
+
+  //  Line Chart
+  const handleLineChart = () => {
+    let dataArray: any ;
+    setBarData("");
+    setPieData("");
+    if(lineData == ''){
+      dataArray = [
+        {"x": 0, "y": 28, "c":0},
+        {"x": 1, "y": 43, "c":0},
+        {"x": 2, "y": 81, "c":0},
+        {"x": 3, "y": 19, "c":0},
+        {"x": 4, "y": 52, "c":0},
+        {"x": 5, "y": 24, "c":0},
+        {"x": 6, "y": 87, "c":0},
+        {"x": 7, "y": 17, "c":0}
+      ];
+      setLineData(JSON.stringify(dataArray, undefined, 1));
+      dataArray = JSON.stringify(dataArray, undefined, 1)
+    }else{
+      dataArray = lineData;
+    }
+    setGraphDataArray(dataArray);
+
+    specs = {
+      "$schema": "https://vega.github.io/schema/vega/v5.json",
+      "description": "A basic line chart example.",
+      "width": 400,
+      "height": 400,
+      "padding": 5,
+      "signals": [
+        { 
+          "name": "interpolate",
+          "value": "linear",
+        }
+      ],    
+      "data": [
+        {
+          "name": "table",
+          "values": dataArray
+        }
```

Comment:
I'd like this spec to be editable too, not just the data. Let's have this be part of the editing pane on the left hand side!